python/ray/autoscaler/gcp/node_provider.py

```diff
@@ -48,7 +48,7 @@ def __init__(self, provider_config, cluster_name):
 
         # Cache of node objects from the last nodes() call. This avoids
         # excessive DescribeInstances requests.
-        self.cached_nodes = {}
+        self.provider_cache = provider_cache or NodeProviderCache()
```

Comment:
Can you remove all the assignments to self.provider_cache inside the custom nodeproviders and only assign it once inside NodeProvider class? e.g.:
`NodeProvider.__init__(self, provider_config, cluster_name, provider_cache)`